### PR TITLE
DRES v2.0.0

### PR DESCRIPTION
**Pull request by sauterl**
This PR elevates DRES v2.0.0-RC4 to v2.0.0

Version 2 of DRES includes major breaking changes, including but not limited to the following:

* Replaced MapDB with Xodus as database layer. Results in no backwards compatibility.
* Entirely rewritten data model enables far more flexible evaluations.
* Terminology change to be inclusive towards non-competition styled evaluation (Competition -> Evaluation Template, Competition Run -> Evaluation (Run)).
* New GUI for evaluation template builder (formerly known as competition builder).
* New, versatile submission endpoint. See corresponding OpenAPI documentation.
* Removed old, GET-based submission endpoint.
* Various bits and pieces that improve performance and stability

### CODE REVIEW

File: backend/src/main/kotlin/dev/dres/DRES.kt [hunk 1]
```diff
@@ -42,7 +47,7 @@ import kotlin.system.exitProcess
  */
 object DRES {
     /** Version of DRES. */
-    const val VERSION = "2.0.0-RC4"
+    const val VERSION = "2.0.0-RC5"
```
Comment (lucaro) [inline backend/src/main/kotlin/dev/dres/DRES.kt:50]: should not be RC5

File: backend/src/main/kotlin/dev/dres/DRES.kt [hunk 2]
```diff
@@ -29,6 +30,10 @@ import kotlinx.dnq.util.initMetaData
 import java.io.File
 import java.nio.file.Path
 import java.nio.file.Paths
+import kotlin.io.path.exists
+import kotlin.io.path.isDirectory
+import kotlin.io.path.isExecutable
+import kotlin.io.path.listDirectoryEntries
```
Comment (lucaro) [inline backend/src/main/kotlin/dev/dres/DRES.kt:36]: what are these for?

File: backend/src/main/kotlin/dev/dres/api/rest/ClientSwaggerPlugin.kt [hunk 3]
```diff
@@ -1,27 +1,37 @@
...
+//            tagsSorter = "'alpha'",
+//            operationsSorter = "'alpha'",
+//            customJavaScriptFiles = emptyList(),
+//            customStylesheetFiles = emptyList()
+//        )
+//
+//        config.router.apiBuilder {
+//            get("/swagger-client", swaggerHandler)
+//        }
+
```
Comment (lucaro) [inline backend/src/main/kotlin/dev/dres/api/rest/ClientSwaggerPlugin.kt:33]: can this be deleted?

File: backend/src/main/kotlin/dev/dres/api/rest/handler/evaluation/scores/TeamGroupScoreHandler.kt [hunk 4]
```diff
@@ -44,7 +44,7 @@ class TeamGroupScoreHandler : AbstractScoreHandler(), GetRestHandler<List<ApiTea
         }
 
 
-            val rac = ctx.runActionContext()
+//            val rac = ctx.runActionContext()
             /* TODO: Not suite sure where the teamGroupAggregator got lost.*/
```
Comment (lucaro) [inline backend/src/main/kotlin/dev/dres/api/rest/handler/evaluation/scores/TeamGroupScoreHandler.kt:48]: what about this?

**Review by lucaro — CHANGES_REQUESTED** [review]
some minor cleanup required

File: backend/src/main/kotlin/dev/dres/DRES.kt (hunk 1 above)
Comment (sauterl) [reply backend/src/main/kotlin/dev/dres/DRES.kt:50]: Good catch, thanks!

File: backend/src/main/kotlin/dev/dres/DRES.kt (hunk 2 above)
Comment (sauterl) [reply backend/src/main/kotlin/dev/dres/DRES.kt:36]: In the first version, ffmpeg checks were implemented here, I removed them.

File: backend/src/main/kotlin/dev/dres/api/rest/ClientSwaggerPlugin.kt (hunk 3 above)
Comment (sauterl) [reply backend/src/main/kotlin/dev/dres/api/rest/ClientSwaggerPlugin.kt:33]: Yes! The Swagger plugin finally enables multiple endpoints

File: backend/src/main/kotlin/dev/dres/api/rest/handler/evaluation/scores/TeamGroupScoreHandler.kt (hunk 4 above)
Comment (sauterl) [reply backend/src/main/kotlin/dev/dres/api/rest/handler/evaluation/scores/TeamGroupScoreHandler.kt:48]: I'll investigate...

Comment (sauterl) [reply backend/src/main/kotlin/dev/dres/api/rest/handler/evaluation/scores/TeamGroupScoreHandler.kt]: Fixed in [c8eedfb](https://github.com/dres-dev/DRES/pull/460/commits/c8eedfba9b3ea084da93e29ebdde0022986577de)

**Review by lucaro — APPROVED** [review]

**Pull request merged**